src/pudl/output/ferc1.py

```diff
@@ -2774,3 +2775,239 @@ def nodes_to_df(calc_forest: nx.DiGraph, nodes: list[NodeId]) -> pd.DataFrame:
+            existing_tags = nx.get_node_attributes(annotated_forest, "tags")
+            node_tags = {
+                node: {"tags": {tag_name: new_node_tag} | existing_tags.get(node, {})}
+            }
+            nx.set_node_attributes(annotated_forest, node_tags)
+            tag = new_node_tag
+        # elif the children disagree then the node's tag shouldn't be set and
+        # the og null tag should be returned
+    return tag
+
+
+def _propagate_tags_to_corrections(annotated_forest: nx.DiGraph) -> nx.DiGraph:
+    existing_tags = nx.get_node_attributes(annotated_forest, "tags")
+    correction_nodes = [
+        node for node in annotated_forest if node.xbrl_factoid.endswith("_correction")
+    ]
+    correction_tags = {}
+    for correction_node in correction_nodes:
+        # for every correction node, we assume that that nodes parent tags can apply
+        parents = list(annotated_forest.predecessors(correction_node))
+        # all correction records shoul have a parent and only one
+        assert len(parents) == 1
+        parent = parents[0]
+        correction_tags[correction_node] = {
+            "tags": existing_tags.get(parent, {})
+            | existing_tags.get(correction_node, {})
+        }
+    nx.set_node_attributes(annotated_forest, correction_tags)
+    return annotated_forest
+
+
+def check_tag_propagation_compared_to_compiled_tags(
+    df: pd.DataFrame,
+    propogated_tag: Literal["in_rate_base"],
+    _out_ferc1__explosion_tags: pd.DataFrame,
+):
+    """Check if tags got propagated.
+
+    Args:
+        df: table to check. This should be either the
+            :func:`out_ferc1__yearly_rate_base`, ``exploded_balance_sheet_assets_ferc1``
+            or ``exploded_balance_sheet_liabilities_ferc1``. The
+            ``exploded_income_statement_ferc1`` table does not currently have propagated
+            tags.
+        propogated_tag: name of tag. Currently ``in_rate_base`` is the only propagated tag.
+        _out_ferc1__explosion_tags: mannually compiled tags. This table includes tags from
+            many of the explosion tables so we will filter it before checking if the tag was
+            propagated.
+
+    Raises:
+        AssertionError: If there are more mannually compiled tags for the ``xbrl_factoids``
+            in ``df`` than found in ``_out_ferc1__explosion_tags``.
+        AssertionError: If there are more mannually compiled tags for the correction
+            ``xbrl_factoids`` in ``df`` than found in ``_out_ferc1__explosion_tags``.
+    """
+    # the tag df has all tags - not just those in a specific explosion
+    # so we need to drop
+    node_idx = list(NodeId._fields)
+    df_filtered = df.filter(node_idx).drop_duplicates()
+    df_tags = _out_ferc1__explosion_tags.merge(
+        df_filtered, on=list(df_filtered.columns), how="right"
+    )
+    mannually_tagged = df_tags[df_tags[propogated_tag].notnull()].xbrl_factoid.unique()
```

Comment:
Couple typos here - `manually_tagged` and `propagated_tag`

Reply:
lol ty